src/domain/event/utils.ts

```diff
@@ -1225,115 +1253,263 @@ export const getRelatedEvents = async ({
-      return true;
-    case EVENT_ACTION_BUTTONS.DELETE:
-      return true;
-    case EVENT_ACTION_BUTTONS.EDIT:
+  const { isDraft, isPublic } = getEventFields(event, 'fi');
+
+  switch (action) {
+    case EVENT_EDIT_ACTIONS.CANCEL:
+    case EVENT_EDIT_ACTIONS.COPY:
+    case EVENT_EDIT_ACTIONS.DELETE:
+    case EVENT_EDIT_ACTIONS.EDIT:
+    case EVENT_EDIT_ACTIONS.POSTPONE:
       return true;
-    case EVENT_ACTION_BUTTONS.POSTPONE:
-      return isPublic;
-    case EVENT_ACTION_BUTTONS.PUBLISH:
-      return isDraft;
-    case EVENT_ACTION_BUTTONS.UPDATE_DRAFT:
+    case EVENT_EDIT_ACTIONS.PUBLISH:
+      return (
+        isDraft &&
+        checkCanUserDoAction({ action, event, organizationAncestors, user })
+      );
+    case EVENT_EDIT_ACTIONS.UPDATE_DRAFT:
       return isDraft;
-    case EVENT_ACTION_BUTTONS.UPDATE_PUBLIC:
+    case EVENT_EDIT_ACTIONS.UPDATE_PUBLIC:
       return isPublic;
   }
 };
 
-const getActionButtonTitle = ({
+export const getEditEventWarning = ({
+  action,
   authenticated,
-  button,
-  isCancelled,
+  event,
   t,
+  userCanDoAction,
 }: {
+  action: EVENT_EDIT_ACTIONS;
   authenticated: boolean;
-  button: EVENT_ACTION_BUTTONS;
-  isCancelled: boolean;
+  event: EventFieldsFragment;
   t: TFunction;
+  userCanDoAction: boolean;
 }): string => {
-  if (AUHENTICATION_NOT_NEEDED.includes(button)) {
+  const { deleted, endTime, eventStatus, isDraft, startTime } = getEventFields(
+    event,
+    'fi'
+  );
+  const isCancelled = eventStatus === EventStatus.EventCancelled;
+
+  const isSubEvent = Boolean(event.superEvent);
+  const eventIsInThePast =
+    (endTime && isPast(endTime)) ||
+    (!endTime && startTime && isBefore(startTime, startOfDay(new Date())));
+
+  if (AUHENTICATION_NOT_NEEDED.includes(action)) {
     return '';
   } else if (!authenticated) {
     return t('authentication.noRightsUpdateEvent');
-  } else if (isCancelled && NOT_ALLOWED_WHEN_CANCELLED.includes(button)) {
-    return t('event.form.editButtonPanel.tooltipCancelledEvent');
+  } else if (isCancelled && NOT_ALLOWED_WHEN_CANCELLED.includes(action)) {
+    return t('event.form.editButtonPanel.warningCancelledEvent');
+  } else if (deleted && NOT_ALLOWED_WHEN_DELETED.includes(action)) {
+    return t('event.form.editButtonPanel.warningDeletedEvent');
+  } else if (eventIsInThePast && NOT_ALLOWED_WHEN_IN_PAST.includes(action)) {
+    return t('event.form.editButtonPanel.warningEventInPast');
+  } else if (isDraft && action === EVENT_EDIT_ACTIONS.CANCEL) {
+    return t('event.form.editButtonPanel.warningCannotCancelDraft');
+  } else if (isDraft && action === EVENT_EDIT_ACTIONS.POSTPONE) {
+    return t('event.form.editButtonPanel.warningCannotPostponeDraft');
+  } else if (isDraft && action === EVENT_EDIT_ACTIONS.PUBLISH && isSubEvent) {
+    return t('event.form.editButtonPanel.warningCannotPublishSubEvent');
+  } else if (!userCanDoAction) {
+    return t('event.form.editButtonPanel.warningNoRightsToEdit');
```

Comment:
Because early return is used here I think `else if` is little bit redundant. I think if only if was used  and with little bit of spacing this would be much easier to read.

Reply:
Fixed